test-skripsi/test-prod.tsx

```diff
@@ -10,27 +17,33 @@
 
 	constructor() {
 		this.inventory = [];
+		this.loadInventory();
+	}
+
+	private findProductById(productId: number): Product | undefined {
+		return this.inventory.find((p) => p.id === productId);
 	}
 
-	addProduct(product: Product) {
-		const index = this.inventory.findIndex((p) => p.id === product.id);
-		if (index !== -1) {
-			console.log("Product already exists.");
-			return;
+	addProduct(product: Product): boolean {
+		if (this.findProductById(product.id)) {
+			throw new InventoryError("Product already exists.");
 		}
 		this.inventory.push(product);
+		this.saveInventory();
+		return true;
 	}
 
-	updateProduct(productId: number, newQuantity: number) {
-		const product = this.inventory.find((p) => p.id === productId);
+	updateProduct(productId: number, changes: Partial<Product>): boolean {
+		const product = this.findProductById(productId);
```

Comment:
When updating a product, consider validating the changes to ensure they are valid for the product's state to prevent invalid data entries.